messaging/src/test/java/org/axonframework/eventhandling/scheduling/dbscheduler/BinaryDbSchedulerEventSchedulerTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.scheduling.dbscheduler;
+
+import com.github.kagkarlsson.scheduler.task.Execution;
+import com.github.kagkarlsson.scheduler.task.Task;
+import com.github.kagkarlsson.scheduler.task.TaskInstance;
+import org.junit.jupiter.api.*;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+
+import static org.awaitility.Awaitility.await;
+import static org.axonframework.utils.DbSchedulerTestUtil.getAndStartScheduler;
+import static org.junit.jupiter.api.Assertions.*;
+
+class BinaryDbSchedulerEventSchedulerTest extends AbstractDbSchedulerEventSchedulerTest {
+
+
+    @Override
+    Task<?> getTask() {
+        return DbSchedulerEventScheduler.binaryTask();
+    }
+
+    @Override
+    boolean useBinaryPojo() {
+        return true;
+    }
+
+    @Test
+    void whenNotInitializedThrow() {
+        eventScheduler.shutdown();
+        scheduler = getAndStartScheduler(dataSource, getTask());
+        try {
+            TaskInstance<DbSchedulerBinaryEventData> instance =
+                    DbSchedulerEventScheduler.binaryTask()
+                                             .instance("id", new DbSchedulerBinaryEventData());
+            scheduler.schedule(instance, Instant.now());
+            await().atMost(Duration.ofSeconds(1L)).untilAsserted(
+                    () -> {
+                        List<Execution> failures = scheduler.getFailingExecutions(Duration.ofHours(1L));
```

Comment:
Does it make sense to validate the thrown exception as our custom exception?

Reply:
It's unlikely it's something else. But I don't know an easy way to test this, as the kind of error is not stored. And the invocation itself is from the `Scheduler`. It likely can be done by putting a spy within the `Scheduler` with reflection, but I don't think that's worth it?